zowe/zowe-explorer-vscode
# fix: Prompt for credentials when 401 in editor, fix profile propagation

## Proposed changes

Fixes #3197, #3289 

### How to test

- Checkout this branch in the repo with VS Code opened
- `pnpm install && pnpm build`
- After the build has finished, click on the Run & Debug panel on the VS Code Side Bar
- Select "Run VS Code Extension" from the tasks list at the top and select the Play ▶ button
- Once opened, make sure that you have (or set) valid credentials on a profile in ZE
- Execute a search on that profile and expand a folder so that a file/PDS member/spool item is visible
- Before opening the file, right click on your profile -> "Manage Profile" -> "Update Credentials"
- Enter in invalid credentials. I recommend using a fake username to prevent lockouts
- Once finished, try to open the file. Notice the error in the editor appears but you are prompted to update your credentials.
- Enter in the valid credentials
- Once done, click "Try again" in the editor to fetch & open the file

## Release Notes




Milestone: 3.0.3?

Changelog:

- Fixed issue where users were not prompted to enter credentials if a 401 error was encountered when opening files, data sets or spools in the editor. [#3197](https://github.com/zowe/zowe-explorer-vscode/issues/3197)
- Fixed issue where profile credential updates or token changes were not reflected within the filesystem. [#3289](https://github.com/zowe/zowe-explorer-vscode/issues/3289)

## Types of changes




## Checklist



**General**

  - **v2 and below:** `yarn workspace vscode-extension-for-zowe vscode:prepublish`
  - **v3:** `pnpm --filter vscode-extension-for-zowe vscode:prepublish`

**Code coverage**


**Deployment**

  - If you're an outside contributor, please post in the [#zowe-doc Slack channel](https://openmainframeproject.slack.com/archives/CC961JYMQ) to coordinate documentation.
  - Otherwise, please check with the rest of the squad about any needed documentation before merging.
